### PR TITLE
Support for job errors in bulk check_jobs function.

### PR DESCRIPTION
**Pull request by rsutormin**

### CODE REVIEW

**Comment by rsutormin** [comment]
@MrCreosote: PR is ready for review.

File: src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java [hunk 1]
```diff
@@ -681,13 +684,41 @@ public static JobState checkJob(String jobId, AuthToken authPart,
...
+	    Map<String, JsonRpcError> checkError = new LinkedHashMap<String, JsonRpcError>();
+	    CheckJobsResults ret = new CheckJobsResults().withJobStates(
+	            new LinkedHashMap<String, JobState>()).withCheckError(checkError);
+	    for (String jobId : params.getJobIds()) {
+	        try {
+	            ret.getJobStates().put(jobId, checkJob(jobId, auth, config));
+	            if (withJobParams) {
+	                jobParams.put(jobId, getJobInputParams(jobId, auth, config, null));
+	            }
+	        } catch (Exception ex) {
```
Comment (MrCreosote) [inline src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java]: This seems like a backwards incompatible change - I assume that this method used to throw an exception if one check failed? Is not throwing an exception going to break code that assumes an exception will be thrown if one of the job check fails?

File: NJSWrapper.spec [hunk 2]
```diff
@@ -241,6 +241,7 @@ module NarrativeJobService {
     typedef structure {
         mapping<job_id, JobState> job_states;
         mapping<job_id, RunJobParams> job_params;
+        mapping<job_id, JsonRpcError> check_error;
```
Comment (MrCreosote) [inline NJSWrapper.spec]: Should probably add docs explaining what this is

File: src/us/kbase/narrativejobservice/test/AweClientDockerJobScriptTest.java [hunk 3]
```diff
@@ -1278,6 +1280,53 @@ public void testLocalSdkCallback() throws Exception {
...
+                    System.out.println(ex.getData());
+                    throw ex;
+                }
+            }
+            String errMsg = "Unexpected app state: " + UObject.getMapper().writeValueAsString(ret);
+            UObject obj = checkJobOutput(ret);
+            List<String> data = obj.asClassInstance(List.class);
+            Assert.assertEquals(errMsg, inputText, data.get(0));
+            Assert.assertEquals(errMsg, "OK", data.get(1));
+            Assert.assertNotNull(hiddenJobId);
```
Comment (MrCreosote) [inline src/us/kbase/narrativejobservice/test/AweClientDockerJobScriptTest.java]: Why would hiddenjobId ever be null? It's assigned once above AFAICT.

File: src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java (hunk 1 above)
Comment (rsutormin) [inline src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java]: Nobody is using it except Narrative. And in Narrative we have a lot of problems with "method used to throw an exception if one check failed". So we DO want to make "backwards incompatible change" and fix bulk job state pulling in Narrative.

File: NJSWrapper.spec (hunk 2 above)
Comment (rsutormin) [inline NJSWrapper.spec]: Good point.

File: src/us/kbase/narrativejobservice/test/AweClientDockerJobScriptTest.java (hunk 3 above)
Comment (rsutormin) [inline src/us/kbase/narrativejobservice/test/AweClientDockerJobScriptTest.java]: Oh shoot...

**Comment by MrCreosote** [comment]
Tests pass, code looks ok

**Pull request merged**